src/barril/units/_scalar.py

```diff
@@ -242,8 +242,8 @@ def GetFormatted(self, unit=None, value_format=None):
     def __eq__(self, other):
         return (
             type(self) is type(other)
-            and self._value == other.value
-            and self._quantity == other._quantity
+            and self._value == other.GetValue(self.unit)
```

Comment:
Due to floating point rounding `self._value == other.GetValue(self.unit)` this could lead to situations where `(a == b) != (b == a)`.
I think this should agree on the unit used for comparison. Maybe the one composed by composing quantities base units.

Reply:
I've refactored to do the comparison always using `DefaultUnit`:

```python
    def __eq__(self, other):
        return (
            type(self) is type(other)
            and self._quantity._category == other._quantity._category
            and self._GetValueInDefaultUnit() == other._GetValueInDefaultUnit()
        )

    def _GetValueInDefaultUnit(self):
        try:
            return self.GetValue(
                self._unit_database.GetDefaultUnit(self._quantity._category)
            )
        except InvalidQuantityTypeError:
            return self._value
```

Reply:
Scalars with the same category could have different units...
I don't think returning the raw value is a good option in `InvalidQuantityTypeError`.

Maybe `return (self._value, self._quantity._unit)`? If you do something like this I recommend a docstring stating that.

Reply:
hmm, ok. I've took you suggestion, but with a slightly modification, take a look.